lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -4,7 +4,9 @@ class Attributes < Base
       def serializable_hash(options = nil)
         options = serialization_options(options)
         options[:fields] ||= instance_options[:fields]
-        serializer.serializable_hash(instance_options, options, self)
+        serialized_hash = serializer.serializable_hash(instance_options, options, self)
```

Comment:
The problem was that since there is no root key for the Attributes adapter, this `serialized_hash` could actually be an array, which was not handled by the `KeyTransform` functions.